examples/so_vits_svc.py

```diff
@@ -0,0 +1,834 @@
+      if i == 0:
+        if self.mode == "group_norm_masked":
+          if padding_mask is not None:
+            _, k, stride = self.cl
+            lengths_org = tilde(padding_mask.cast(dtypes.bool)).cast(dtypes.int64).sum(1)  # ensure padding_mask is bool for tilde
+            lengths = (((lengths_org - k) / stride) + 1).floor().cast(dtypes.int64)
+            padding_mask = tilde(lengths_to_padding_mask(lengths)).cast(dtypes.int64)  # lengths_to_padding_mask returns bool tensor
+          x = conv(x, padding_mask)  # padding_mask is numeric
+        else:
+          x = conv(x)  # default
+      else:
+        x = conv(x)
+    return x
+
+# helpers:
+def randn_like(x): return Tensor.randn(*x.shape, dtype=x.dtype).to(device=x.device)
+
+def tilde(x: Tensor) -> Tensor:
+  if x.dtype == dtypes.bool: return (1 - x).cast(dtypes.bool)
+  return (x + 1) * -1  # this seems to be what the ~ operator does in pytorch for non bool
+
+def lengths_to_padding_mask(lens: Tensor) -> Tensor:
+  bsz, max_lens = lens.shape[0], lens.max().numpy().item()
+  mask = Tensor.arange(max_lens).to(lens.device).reshape(1, max_lens)
+  mask = mask.expand(bsz, -1) >= lens.reshape(bsz, 1).expand(-1, max_lens)
+  return mask.cast(dtypes.bool)
+
+class Sequential:  # allows recursive sequential
+  def __init__(self, list): self.list=list
+  def __call__(self, x: Tensor): return x.sequential(self.list)
+  def __getitem__(self, i: int): return self.list[i]  # for loading weights
+
+class Dropout:
+  def __init__(self, p=0.5): self.p=p
+  def __call__(self, x: Tensor): return x.dropout(self.p)
+
+class GELU:
+  def __init__(self): pass
+  def __call__(self, x: Tensor): return x.gelu()
+
+class SamePad:
+  def __init__(self, kernel_size, causal=False):
+    self.remove = (kernel_size - 1) if causal else (1 if kernel_size % 2 == 0 else 0)
+  def __call__(self, x):
+    if self.remove > 0:
+      x = x[:, :, : -self.remove]
+    return x
+
+class TransposeLast:
+  def __init__(self, deconstruct_idx=None): self.deconstruct_idx = deconstruct_idx
+  def __call__(self, x: Tensor):
+    if self.deconstruct_idx is not None:
+      x = x[self.deconstruct_idx]
+    return x.transpose(-2, -1)
+
+class Fp32LayerNorm(nn.LayerNorm):
+  def __init__(self, *args, **kwargs): super().__init__(*args, **kwargs)
+  def __call__(self, input: Tensor):
+    self.weight = self.weight.float() if self.weight is not None else None
+    self.bias = self.bias.float() if self.bias is not None else None
+    output = super().__call__(input.float())
+    return output.cast(input.dtype)
+
+class Fp32GroupNorm(nn.GroupNorm):
+  def __init__(self, *args, **kwargs):
+    super().__init__(*args, **kwargs)
+  def __call__(self, input: Tensor):
+    self.weight = self.weight.float() if self.weight is not None else None
+    self.bias = self.bias.float() if self.bias is not None else None
+    output = super().__call__(input.float())
+    return output.cast(input.dtype)
```

Comment:
Can you just cast while loading the weights? instead of making new classes.

Reply:
[75f6c22](https://github.com/tinygrad/tinygrad/pull/1371/commits/75f6c22c87efebfd5bc1754cb594e4bd3f9aa43f)